sys_util/src/struct_util.rs

```diff
@@ -23,9 +33,9 @@ pub type Result<T> = std::result::Result<T, Error>;
 ///
 /// * `f` - The input to read from.  Often this is a file.
 /// * `out` - The struct to fill with data read from `f`.
-pub unsafe fn read_struct<T: Copy, F: Read>(f: &mut F, out: &mut T) -> Result<()> {
+pub unsafe fn read_struct<T: Copy, F: io::Read>(f: &mut F, out: &mut T) -> Result<()> {
     let out_slice = std::slice::from_raw_parts_mut(out as *mut T as *mut u8, mem::size_of::<T>());
-    f.read_exact(out_slice).map_err(|_| Error::ReadStruct)?;
+    f.read_exact(out_slice).map_err(|e| Error::ReadStruct(e))?;
```

Comment:
This can be simplified: ``` f.read_exact(out_slice).map_err(Error::ReadStruct)?;```. That's why the integration tests fail as well.

Checkout the clippy explanation: https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure

Same applies for the other `map_err`.

Reply:
@andreeaflorescu , thanks so much for your pointing. I updated this and all checks seem to be passed. PTAL.